firefox.sh

```diff
@@ -1,3 +1,20 @@
 #!/bin/sh
 
-exec /app/extra/bin/vlc
+JRE_PLUGIN_FILENAME="libnpjp2.so"
+
+mkdir -p $HOME/.mozilla/plugins
+mkdir -p $HOME/.flatpak_extras/firefox
+
+if [ ! -f $HOME/.mozilla/plugins/$JRE_PLUGIN_FILENAME ]; then
+	wget --no-cookies --no-check-certificate --header "Cookie: gpw_e24=http%3A%2F%2Fwww.oracle.com%2F; oraclelicense=accept-securebackup-cookie" "http://download.oracle.com/otn-pub/java/jdk/8u112-b15/jre-8u112-linux-x64.tar.gz"
+	tar zxf jre-8u112-linux-x64.tar.gz
+	cp -r jre1.8.0_112/{bin,lib,man,plugin} $HOME/.flatpak_extras/firefox
+	ln -s $HOME/.flatpak_extras/firefox/lib/amd64/libnpjp2.so $HOME/.mozilla/plugins/$JRE_PLUGIN_FILENAME
+	rm -rf jre1.8.0_112 jre-8u112-linux.x64.tar-gz
```

Comment:
You can't install it when applying the extra_data!

Unfortunately, to download anything related to Oracle Java, you need to provide a custom cookie (that's a workaround for agreeing to their license). Flatpak cannot add custom cookies (and probably would have problems with 302 HTTP code as well), so the only option is to use wget on the first run and keep all JRE-related files in homedir (which is mounted from host).